packages/grafana-data/src/types/data.ts

```diff
@@ -15,7 +15,7 @@ export enum LoadingState {
   Error = 'Error',
 }
 
-export type PreferredVisualisationType = 'graph' | 'table';
+export type PreferredVisualisationType = 'graph' | 'table' | 'logs' | 'trace';
```

Comment:
I think we can use this to figure out where to show the dataFrames. For data sources which won't use that we should just make sure the output is sensible (ie fallback to table table/graph)

Reply:
:+1: